var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -141,4 +163,19 @@ def install(self, spec, prefix):
         install_tree('lib', prefix.lib)
         install_tree('include', prefix.include)
         if '~mpi' in spec:
-            install('libseq/libmpiseq.a', prefix.lib)
+            lib_dsuffix = '.dylib' if sys.platform == 'darwin' else '.so'
+            lib_suffix = lib_dsuffix if '+shared' in spec else '.a'
+            install('libseq/libmpiseq%s' % lib_suffix, prefix.lib)
+
+        if '~mpi' in spec: # FIXME: A process or daemon was unable to complete a TCP connection to another process
+            with working_dir('examples', create=False):
+                mpirun = join_path(self.spec['mpi'].prefix.bin, 'mpirun') # FIXME: avoid hardcoding mpirun (could be mpiexec, mpiexec.hydra, etc)
```

Comment:
I'm more impressed that the python code will work at all in this case, I would have guessed that with `~mpi`, spec['mpi'] would not be defined.

But your reply was perhaps targeted at my first comment that I retracted after seeing the 'hidden' discussion on the `variant`